Library/Homebrew/tap.rb

```diff
@@ -280,4 +316,93 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
```

Comment:
I still don't like that this is inheriting from `Tap` and called a tap when it is not a tap in any sense. If you really, really don't want to use a `Module` can you remove `Tap` from the name (it is not currently a real `Tap`) e.g. `CoreFormulaRepository` and add documentation explaining why it inherits from `Tap` and what the future plans are which will make that inheritance make more sense.